### PR TITLE
Improve comments for OutgoingResponseFrame forwarding constructor

### PR DESCRIPTION
**Pull request by pepone**
This small PR just improve the comments in the OutgoingResponseFrame forwarding constructor to make clear what it does.

### CODE REVIEW

File: csharp/src/Ice/OutgoingResponseFrame.cs [hunk 1]
```diff
@@ -174,6 +182,11 @@ public static OutgoingResponseFrame WithVoidReturnValue(Current current)
                         var replyStatus = (ReplyStatus)response.Payload[0];
                         if (replyStatus == ReplyStatus.UserException)
                         {
+                            // We are forwarding an ice1 user exception using ice2, we create an ice2 encapsulation
+                            // and write the 1.1 reply status followed by the user exception. The size of the new
+                            // encapsulation is adjusted, -1 correspond to the result type that is included in the
+                            // payload but is not part of the encapsulation, +1 correspond to the 1.1 replyStatus
+                            // that is included at the start of the encapsulation.
```
Comment (externl) [inline csharp/src/Ice/OutgoingResponseFrame.cs:189]:
```suggestion
                            // We are forwarding an ice1 user exception using ice2. We create an ice2 encapsulation
                            // and write the 1.1 reply status followed by the user exception. The size of the new
                            // encapsulation is adjusted: -1 corresponds to a result type that is included in the
                            // payload but is not part of the encapsulation; +1 corresponds to a 1.1 replyStatus
                            // that is included at the start of the encapsulation.
```

File: csharp/src/Ice/OutgoingResponseFrame.cs [hunk 2]
```diff
@@ -186,11 +199,15 @@ public static OutgoingResponseFrame WithVoidReturnValue(Current current)
                         }
                         else
                         {
+                            // We are forwarding an ice1 system exception using ice2, we create an ice2 encapsulation
+                            // and write the 1.1 encoded exception in it (which includes the 1.1 reply status, followed
+                            // by the exception). The size of the new encapsulation is the size of the
+                            // payload +2 bytes for the encapsulations encoding.
```
Comment (externl) [inline csharp/src/Ice/OutgoingResponseFrame.cs:205]:
```suggestion
                            // We are forwarding an ice1 system exception using ice2.  We create an ice2 encapsulation
                            // and write the 1.1 encoded exception in it (which includes the 1.1 reply status, followed
                            // by the exception). The size of the new encapsulation is the size of the
                            // payload +2 bytes for the encapsulations encoding.
```

**Review by externl — APPROVED** [review]

File: csharp/src/Ice/OutgoingResponseFrame.cs [hunk 3]
```diff
@@ -142,14 +143,15 @@ public static OutgoingResponseFrame WithVoidReturnValue(Current current)
                     {
                         Debug.Assert(response.Protocol == Protocol.Ice2);
 
-                        // Read the reply status byte immediately after the encapsulation header; + 2 corresponds to the
-                        // encoding in the header.
-                        byte b = response.Payload[1 + sizeLength + 2];
-                        ReplyStatus replyStatus = b.AsReplyStatus();
+                        // Read the reply status byte immediately after the encapsulation header; +1 correspond to the
```
Comment (bernardnormier) [inline csharp/src/Ice/OutgoingResponseFrame.cs:146]: corresponds

File: csharp/src/Ice/OutgoingResponseFrame.cs [hunk 4]
```diff
@@ -142,14 +143,15 @@ public static OutgoingResponseFrame WithVoidReturnValue(Current current)
...
+                        // result type and +2 corresponds to the encoding in the encapsulation header.
+                        ReplyStatus replyStatus = response.Payload[1 + sizeLength + 2].AsReplyStatus();
 
-                        buffer[0] = b;
+                        buffer[0] = (byte)replyStatus;
                         if (replyStatus == ReplyStatus.UserException)
                         {
+                            // We are forwarding an user exception encoded using 1.1 encoding and ice2 protocol over
```
Comment (bernardnormier) [inline csharp/src/Ice/OutgoingResponseFrame.cs:153]: Suggestion: We are forwarding a user exception encoded using 1.1 received over ice2 to ice1. Note that we never perform any transcoding.

File: csharp/src/Ice/OutgoingResponseFrame.cs [hunk 5]
```diff
@@ -161,8 +163,14 @@ public static OutgoingResponseFrame WithVoidReturnValue(Current current)
                         }
                         else
                         {
-                            Data[0] = Data[0].Slice(0, 1);
+                            // We are forwarding a system exception encoded using 1.1 encoding and ice2 protocol over
```
Comment (bernardnormier) [inline csharp/src/Ice/OutgoingResponseFrame.cs:166]: We are forwarding a system exception encoded using 1.1 and received over ice2 to ice1.

File: csharp/src/Ice/OutgoingResponseFrame.cs [hunk 6]
```diff
@@ -174,6 +182,11 @@ public static OutgoingResponseFrame WithVoidReturnValue(Current current)
                         var replyStatus = (ReplyStatus)response.Payload[0];
                         if (replyStatus == ReplyStatus.UserException)
                         {
+                            // We are forwarding an ice1 user exception using ice2. We create an ice2 encapsulation
```
Comment (bernardnormier) [inline csharp/src/Ice/OutgoingResponseFrame.cs:185]: For consistency with the previous comments, I would use:
We are forwarding a user exception encoded using 1.1 and received over ice1 to ice2.

File: csharp/src/Ice/OutgoingResponseFrame.cs [hunk 7]
```diff
@@ -174,6 +182,11 @@ public static OutgoingResponseFrame WithVoidReturnValue(Current current)
...
+                            // and write the 1.1 reply status followed by the user exception. The size of the new
```
Comment (bernardnormier) [inline csharp/src/Ice/OutgoingResponseFrame.cs:186]: 1.1 reply status followed by the 1.1-encoded data of the user exception.

File: csharp/src/Ice/OutgoingResponseFrame.cs [hunk 8]
```diff
@@ -174,6 +182,11 @@ public static OutgoingResponseFrame WithVoidReturnValue(Current current)
...
+                            // encapsulation is adjusted: -1 corresponds to a result type that is included in the
+                            // payload but is not part of the encapsulation; +1 corresponds to a 1.1 replyStatus
```
Comment (bernardnormier) [inline csharp/src/Ice/OutgoingResponseFrame.cs:188]: This is not quite correct as we did not receive a result type in the ice1/1.1 payload. Maybe just remove the -1 / +1 below and remove this comment?

File: csharp/src/Ice/OutgoingResponseFrame.cs [hunk 9]
```diff
@@ -186,11 +199,15 @@ public static OutgoingResponseFrame WithVoidReturnValue(Current current)
                         }
                         else
                         {
+                            // We are forwarding an ice1 system exception using ice2.  We create an ice2 encapsulation
```
Comment (bernardnormier) [inline csharp/src/Ice/OutgoingResponseFrame.cs:202]: For consistency with the previous comments, I would use: We are forwarding a system exception encoded using 1.1 and received over ice1 to ice2.

File: csharp/src/Ice/OutgoingResponseFrame.cs [hunk 10]
```diff
@@ -199,6 +216,9 @@ public static OutgoingResponseFrame WithVoidReturnValue(Current current)
                 }
                 else
                 {
+                    // In this case we always have an encapsulation it is either an ice1 response with result type
```
Comment (bernardnormier) [inline csharp/src/Ice/OutgoingResponseFrame.cs:219]: An ice1 response or reply has ReplyStatus OK, which has the same value as an ice2 response with ResultType Success.
And here, the only thing that changes is the encapsulation header. "Rewrite with the new encoding" sounds like we're transcoding.

File: csharp/src/Ice/OutgoingResponseFrame.cs [hunk 11]
```diff
@@ -210,7 +230,9 @@ public static OutgoingResponseFrame WithVoidReturnValue(Current current)
                     Data.Add(response.Payload.Slice(1 + sizeLength + 2));
                 }
 
-                // There is never a binary context in this case.
+                // When forwarding responses using different protocols there is never a binary context. Either the
+                // source response uses ice1 and has not a binary context or the source response uses ice2 and its
```
Comment (bernardnormier) [inline csharp/src/Ice/OutgoingResponseFrame.cs:234]: ... and does not carry a binary context ...

**Review by bernardnormier — COMMENTED** [review]
Added a few suggestions.

File: csharp/src/Ice/OutgoingResponseFrame.cs (hunk 3 above)
Comment (pepone) [reply csharp/src/Ice/OutgoingResponseFrame.cs]: fixed

**Comment by pepone** [comment]
Pushed a few more updates to comments according to review comments

**Review by bernardnormier — APPROVED** [review]
Looks good.

**Pull request merged**